src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -3703,13 +3703,146 @@ _test_sample_versioned_api_example_4 (void)
+   ASSERT_OR_PRINT (ok, error);
+   db = mongoc_client_get_database (client, "db");
+   sales = mongoc_database_get_collection (db, "sales");
+   /* Drop db.sales in case the collection exists. */
+   ok = mongoc_collection_drop (sales, &error);
+   if (!ok && NULL == strstr (error.message, "ns not found")) {
+      /* Ignore an "ns not found" error on dropping the collection in case the
+       * namespace does not exist. */
+      ASSERT_OR_PRINT (ok, error);
+   }
+
+   /* Start Versioned API Example 5 */
+   docs[0] = BCON_NEW ("_id", BCON_INT32 (1),
+                       "item", "abc",
+                       "price", BCON_INT32 (10),
+                       "quantity", BCON_INT32 (2),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-01-01T08:00:00Z")));
+   docs[1] = BCON_NEW ("_id", BCON_INT32 (2),
+                       "item", "jkl",
+                       "price", BCON_INT32 (20),
+                       "quantity", BCON_INT32 (1),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-03T09:00:00Z")));
+   docs[2] = BCON_NEW ("_id", BCON_INT32 (3),
+                       "item", "xyz",
+                       "price", BCON_INT32 (5),
+                       "quantity", BCON_INT32 (5),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-03T09:05:00Z")));
+   docs[3] = BCON_NEW ("_id", BCON_INT32 (4),
+                       "item", "abc",
+                       "price", BCON_INT32 (10),
+                       "quantity", BCON_INT32 (10),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-15T08:00:00Z")));
+   docs[4] = BCON_NEW ("_id", BCON_INT32 (5),
+                       "item", "xyz",
+                       "price", BCON_INT32 (5),
+                       "quantity",BCON_INT32 (10),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-15T09:05:00Z")));
+   docs[5] = BCON_NEW ("_id", BCON_INT32 (6),
+                       "item", "xyz",
+                       "price", BCON_INT32 (5),
+                       "quantity",BCON_INT32 (5),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-15T12:05:10Z")));
+   docs[6] = BCON_NEW ("_id", BCON_INT32 (7),
+                       "item", "xyz",
+                       "price", BCON_INT32 (5),
+                       "quantity",BCON_INT32 (10),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-02-15T14:12:12Z")));
+   docs[7] = BCON_NEW ("_id", BCON_INT32 (8),
+                       "item", "abc",
+                       "price", BCON_INT32 (10),
+                       "quantity",BCON_INT32 (5),
+                       "date", BCON_DATE_TIME (iso_to_unix ("2021-03-16T20:20:13Z")));
+   ok = mongoc_collection_insert_many (
+      sales, (const bson_t**) docs, N_DOCS, NULL /* opts */, &reply, &error);
+   /* End Versioned API Example 5 */
+   ASSERT_OR_PRINT (ok, error);
+   bson_destroy (&reply);
```

Comment:
Is "docs", etc. also released?

Reply:
Yes, the individual documents are released in the for loop on L3829.